js/scatter_widget.js

```diff
@@ -202,6 +213,23 @@ function render({ model, el }) {
     update_counts();
   }
 
+  function dragged(event) {
+    isDragging = true;
+    // Add new datapoints to the screen, but not to the datastore just yet
+    let r1 = container.getBoundingClientRect();
+    let r2 = svg.node().getBoundingClientRect();
+    let size = size_input.value;
+    let new_x = event.x + (Math.random() - 0.5) * size;
+    let corrected_y = event.y - (r2.y - r1.y);
+    let new_y = corrected_y + (Math.random() - 0.5) * size;
+    add_point(new_x, new_y);
+    update_brush(event.x, corrected_y);
+  }
+
+  function drag_end(event) {
+    finish_batch();
+  }
+
   function grab_data() {
     // We need to account for the fact that we draw the y-axis other way around in svg land
```

Comment:
BTW this seems to be a half-implemented function... it seems like your idea was to keep the SVG coordinates in the `data` object, and only do the `height - y` conversion inside this function. So basically the `height - ` could be removed from https://github.com/koaning/drawdata/pull/22/files#diff-d002501e4ac8138098cd0c9a93ed325f49da4a4f15a556d75be8824af97a58e3L128 and https://github.com/koaning/drawdata/pull/22/files#diff-d002501e4ac8138098cd0c9a93ed325f49da4a4f15a556d75be8824af97a58e3L192 and instead moved into https://github.com/koaning/drawdata/pull/22/files#diff-d002501e4ac8138098cd0c9a93ed325f49da4a4f15a556d75be8824af97a58e3L207 - let me know if you'd like me to add that in this same PR. :)

Reply:
It really has been a while since I touched the codebase 😅 but I agree that a single click should draw a single point. I'll try to find the time today to confirm that PR works. 

If possible, please make another PR for the other fix.

Reply:
actually, on second thought it's probably better to keep it as is in case someone wants to edit the contents of the `data` attribute from the python side... but we can fix the name/description of `grab_data` to make more sense